crates/ruff_linter/src/rules/pycodestyle/helpers.rs

```diff
@@ -80,27 +83,51 @@ pub(super) fn is_overlong(
         return None;
     }
 
-    let mut width = LineWidthBuilder::new(tab_size);
-    width = width.add_str(line.as_str());
+    // If the line ends in a pragma, remove it from the considered width.
+    let mut line = Cow::Borrowed(line);
+    if let [comment_range] = indexer.comment_ranges().comments_in_range(line.range()) {
```

Comment:
I'm not sure if we want to do this before or after computing the line width. We can avoid this lookup if we do it _after_ computing the line width, but then we need to _recompute_ the width after truncating the line.